models/eprop_synapse.h

```diff
@@ -374,25 +344,144 @@ class eprop_synapse : public Connection< targetidentifierT >
   //! Low-pass filter of the eligibility trace.
   double kappa_;
 
-  //! First moment estimate of Adam optimizer.
-  double adam_m_;
-
-  //! Second moment raw estimate of Adam optimizer.
-  double adam_v_;
-
-  //! Sum over all gradients in one batch.
-  double sum_grads_;
-
   //! If this connection is between two recurrent neurons.
   bool is_recurrent_to_recurrent_conn_;
 
   //! Vector of presynaptic inter-spike-intervals.
   std::vector< long > presyn_isis_;
+
+public:
+  /**
+   *  Optimizer
+   *  Pointer is managed entirely by Connector
+   *   @todo Make non-public!
+   */
+  EpropOptimizer* optimizer_;
 };
 
 template < typename targetidentifierT >
 constexpr ConnectionModelProperties eprop_synapse< targetidentifierT >::properties;
 
+template < typename targetidentifierT >
+eprop_synapse< targetidentifierT >::eprop_synapse()
+  : ConnectionBase()
+  , weight_( 1.0 )
+  , t_previous_spike_( 0 )
+  , t_previous_update_( 0 )
+  , t_next_update_( 0 )
+  , t_previous_trigger_spike_( 0 )
+  , tau_m_readout_( 10.0 )
+  , kappa_( std::exp( -Time::get_resolution().get_ms() / tau_m_readout_ ) )
+  , is_recurrent_to_recurrent_conn_( false )
+  , optimizer_( nullptr )
+{
+  // std::cout << "Create this " << this << ", optimizer " << optimizer_ << std::endl;
+}
+
+template < typename targetidentifierT >
+eprop_synapse< targetidentifierT >::~eprop_synapse()
+{
+  // explicitly do not delete optimizer_, handled by Connector
+  // std::cout << "Delete this " << this << ", optimizer " << optimizer_ << std::endl;
+}
+
+// This copy constructor is used to create instances from prototypes.
+// Therefore, only parameter values are copied.
+template < typename targetidentifierT >
+eprop_synapse< targetidentifierT >::eprop_synapse( const eprop_synapse& es )
+  : ConnectionBase( es )
+  , weight_( es.weight_ )
+  , t_previous_spike_( 0 )
+  , t_previous_update_( 0 )
+  , t_next_update_( kernel().simulation_manager.get_eprop_update_interval().get_steps() )
+  , t_previous_trigger_spike_( 0 )
+  , tau_m_readout_( es.tau_m_readout_ )
+  , kappa_( std::exp( -Time::get_resolution().get_ms() / tau_m_readout_ ) )
+  , is_recurrent_to_recurrent_conn_( es.is_recurrent_to_recurrent_conn_ )
+  , optimizer_( es.optimizer_ )
+{
+  // std::cout << "Copy   this " << this << ", optimizer " << optimizer_ <<
+  // " from this " << &es << ", optimizer " << es.optimizer_ << std::endl;
```

Comment:
Is this a left-behind debugging comment?
```suggestion
  // " from this " << &es << ", optimizer " << es.optimizer_ << std::endl;
```